apm-agent-plugins/apm-api-plugin/src/main/java/co/elastic/apm/agent/pluginapi/AbstractSpanInstrumentation.java

```diff
@@ -132,6 +132,17 @@ public static Object doCreateSpan(@Advice.FieldValue(value = "span", typing = As
         }
     }
 
+    public static class InitializeInstrumentation extends AbstractSpanInstrumentation {
+        public InitializeInstrumentation() {
+            super(named("initialize").and(takesArgument(0, Object.class)));
+        }
+
+        @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
+        public static void incrementReferences(@Advice.Argument(0) Object span) {
+            ((AbstractSpan<?>) span).incrementReferences();
```

Comment:
Have you considered instrumenting `AbstractSpanImpl` directly and incrementing the reference there? The advantage is that would also work with older versions of the API.

Reply:
If you mean the constructor, then - yes, I considered but preferred this.
As far as I see it - there is no problem to any current API user (besides the one that reported this issue) and all future users will use the newer API version.

Reply:
👌